etl/steps/data/garden/who/2023-08-01/mortality_database.py

```diff
@@ -0,0 +1,173 @@
+    """
+    Improve the labelling of the sex column
+    """
+    sex_dict = {"All": "Both sexes", "Female": "Females", "Male": "Males", "Unknown": "Unknown sex"}
+    tb["sex"] = tb["sex"].replace(sex_dict, regex=False)
+    return tb
+
+
+def tidy_age_dimension(tb: Table) -> Table:
+    age_dict = {
+        "[Unknown]": "Unknown age",
+        "[85+]": "over 85 years",
+        "[80-84]": "80-84 years",
+        "[75-79]": "75-49 years",
+        "[70-74]": "70-74 years",
+        "[65-69]": "65-69 years",
+        "[60-64]": "60-64 years",
+        "[55-59]": "55-59 years",
+        "[50-54]": "50-54 years",
+        "[45-49]": "45-49 years",
+        "[40-44]": "40-44 years",
+        "[35-39]": "35-39 years",
+        "[30-34]": "30-34 years",
+        "[25-29]": "25-29 years",
+        "[20-24]": "20-24years",
+        "[15-19]": "15-19 years",
+        "[10-14]": "10-14 years",
+        "[5-9]": "5-9 years",
+        "[1-4]": "1-4 years",
+        "[0]": "less than 1 year",
+        "[All]": "all ages",
+    }
+
+    tb["age_group"] = tb["age_group"].replace(age_dict, regex=False)
+
+    return tb
+
+
+def add_metadata(dest_dir: str, ds_meadow: Dataset, tb: Table) -> Dataset:
+    """
+    Adding metadata at the variable level
+    First step is to group by the dims, which are normally: age, sex and cause.
+    Then for each variable (the different metrics) we add in the metadata.
+    """
+    ds_garden = Dataset.create_empty(dest_dir, metadata=ds_meadow.metadata)
+    dims = ["cause", "sex", "age_group", "icd10_codes", "broad_cause_group"]
+    tb_group = tb.groupby(dims)
+    for group_id, group in tb_group:
+        # Grab out the IDs of each of the grouping factors, e.g. the age-group, sex and cause
+        group = Table(group)
+        dims_id = dict(zip(dims, group_id))
+        # Create the unique table short name
+        keys_to_extract = ["cause", "sex", "age_group"]
+        # Extract values for the specified keys
+        name_list = [dims_id[key] for key in keys_to_extract]
+        group.metadata.short_name = underscore(" - ".join(name_list))[0:240]
+        variables = group.columns.drop(dims + ["country", "year"])
+        for variable_name in variables:
+            group[variable_name] = Variable(group[variable_name])
+            # Create all the necessary metadata
+            cleaned_variable = create_variable_metadata(variable=group[variable_name], **dims_id)
+            group[cleaned_variable.name] = cleaned_variable
+            group = group.drop(columns=variable_name)
+            # dropping columns that are totally empty - not all combinations of variables exist or have been downloaded
+        group = group.dropna(axis=1, how="all")
+        # Dropping dims as table name contains them
+        group = group.drop(columns=dims)
+        group = group.set_index(["country", "year"], verify_integrity=True)
+        ds_garden.add(group)
+    return ds_garden
+
+
+def create_variable_metadata(
```

Comment:
this is super optional and would only be mainly useful if others want to adapt your code but you could add a bit more documentation on how you create variable metadata here (ie. a slightly more extended description on what the function does). but to be honest it's also quite clear once you read it :) so completely up to you!

Reply:
Very good point, the docstring is not very informative at the moment!